File: backend/model/treats.js
const mongoosee = require('mongoose');
const { Schema } = mongoosee;


const TreatsSchema = new Schema({
    treatName: {
        type: String,
        required: true,
    },
    treatDescription: {
        type: String,
        required: false,
    },
    treatPrice: {
        type: Number,
        required: true,
    },
    treatSalePrice : {
        type: Number,
        required: false,
        default: 0
    },
    treatSKU: {
        type: String,
        required: false,
    },
    
    stockStatus: {
        type: String,
        required: true,
        default: 'in-stock'
    },
    treatImage: {
        type: String,
        
    },
});

const Treats = mongoosee.model('Treats', TreatsSchema);

module.exports = Treats;

File: lib/models/treats.dart

/*import 'package:freezed_annotation/freezed_annotation.dart';

part 'treats.freezed.dart';
part 'treats.g.dart';

List<Treats> treatsFromJson(dynamic data) =>
    List<Treats>.from(data.map((x) => Treats.fromJson(x)));

@freezed

abstract class Treats with _$Treats {
  factory Treats(
    {
    Required String treatName,
    Required String treatId,
    Required double treatPrice,
    Required String treatDescription,
    Required String treatSKU,
    Required double treatSalePrice,
    Required String treatImage,
}) = _Treats;

  factory Treats.fromJson(Map<String, dynamic> json) => _$TreatsFromJson(json);
}
extension TreatExt on Treats {
  String get fullImagePath => 
}
```*/

  